tests/unit/providers/__init__.py

```diff
@@ -0,0 +1,55 @@
+from abc import ABC, abstractmethod
+
+import pytest
+
+from rockcraft.providers import Provider
+
+
+class BaseProviderTest(ABC):
```

Comment:
This is the "major" change in the PR: I wanted to reused those two tests in both `test_lxd` and `test_multipass` without getting too "magical" with the fixtures. Eager to hear if there's a cleaner way to do this, though.

Reply:
If this is controversial or not it does not matter as much as this code is moving into craft-providers; correct me if I am wrong @mr-cal